baseplate/diagnostics/metrics.py

```diff
@@ -30,31 +30,53 @@ def __init__(self, client):
         self.client = client
 
     def on_server_span_created(self, context, server_span):
-        context.metrics = self.client.batch()
-        observer = MetricsServerSpanObserver(context.metrics, "server." + server_span.name)
+        batch = self.client.batch()
+        context.metrics = batch
+        observer = MetricsServerSpanObserver(batch, server_span)
         server_span.register(observer)
 
 
-class MetricsSpanObserver(SpanObserver):
-    def __init__(self, batch, name):
+class MetricsServerSpanObserver(SpanObserver):
+    def __init__(self, batch, server_span):
         self.batch = batch
-        self.timer = batch.timer(name)
+        self.timer = batch.timer("server." + server_span.name)
 
     def on_start(self):
         self.timer.start()
 
     def on_finish(self, exc_info):
         self.timer.stop()
+        self.batch.flush()
 
-
-class MetricsServerSpanObserver(MetricsSpanObserver):
     def on_child_span_created(self, span):  # pragma: nocover
         if isinstance(span, LocalSpan):
-            observer = MetricsSpanObserver(self.batch, span.component_name + "." + span.name)
+            observer = MetricsLocalSpanObserver(self.batch, span)
         else:
-            observer = MetricsSpanObserver(self.batch, "clients." + span.name)
+            observer = MetricsClientSpanObserver(self.batch, span)
         span.register(observer)
 
+
+class MetricsLocalSpanObserver(SpanObserver):
+    def __init__(self, batch, span):
+        self.timer = batch.timer(span.component_name + "." + span.name)
+
+    def on_start(self):
+        self.timer.start()
+
     def on_finish(self, exc_info):
-        super(MetricsServerSpanObserver, self).on_finish(exc_info)
-        self.batch.flush()
+        self.timer.stop()
+
+
+class MetricsClientSpanObserver(SpanObserver):
+    def __init__(self, batch, span):
+        self.batch = batch
+        self.base_name = "clients." + span.name
+        self.timer = batch.timer(self.base_name)
+
+    def on_start(self):
+        self.timer.start()
+
+    def on_finish(self, exc_info):
+        self.timer.stop()
+        suffix = "success" if not exc_info else "failure"
+        self.batch.counter(self.base_name + "." + suffix).increment()
```

Comment:
Yup, it'll include it. The span names are up to the various integrations but some examples are:

* [`server.count_multi`](https://github.com/reddit/baseplate/blob/master/baseplate/integration/thrift/__init__.py#L75)
* [`clients.activity_service.count_multi`](https://github.com/reddit/baseplate/blob/master/baseplate/context/thrift.py#L91)
* [`clients.memcache.get`](https://github.com/reddit/baseplate/blob/master/baseplate/context/memcache/__init__.py#L114)